pkg/core/xds/types.go

```diff
@@ -190,10 +189,17 @@ type ZoneEgressProxy struct {
 	MeshResourcesList  []*MeshResources
 }
 
+type MeshIngressResources struct {
+	Mesh        *core_mesh.MeshResource
+	EndpointMap EndpointMap
+}
+
 type ZoneIngressProxy struct {
-	GatewayRoutes   *core_mesh.MeshGatewayRouteResourceList
-	MeshGateways    *core_mesh.MeshGatewayResourceList
-	PolicyResources map[core_model.ResourceType]core_model.ResourceList
+	ZoneIngressResource *core_mesh.ZoneIngressResource
+	GatewayRoutes       *core_mesh.MeshGatewayRouteResourceList
+	MeshGateways        *core_mesh.MeshGatewayResourceList
+	PolicyResources     map[core_model.ResourceType]core_model.ResourceList
+	MeshResourceList    []*MeshIngressResources
```

Comment:
why not have a `map` here? Can we use the same pattern we use for `CrossMeshResources` in the Dataplane?

Reply:
I used the same pattern as in `ZoneEgressProxy`. Probably we don't need `map` because we always iterate over this list when generate things